### PR TITLE
Return entire CODAP message response instead of just response.values/response.error.

### PR DESCRIPTION
**Pull request by lublagg**
The decision to make this change came from working with Evangeline on the `codap-plugin-starter-project`. Initially the `sendMessage` function was copied from the original `codap-plugin-starter-project` code, but we realized that, at least for the purposes of the starter plugin, we wanted to grab the whole response from the `sendRequest` function, not only the resolved values / rejected error, in order to display the entire response message in the UI.

I'm not sure whether or not this is the best approach, as it puts the responsibility to implement error handling on the user of the API. 

Also, there is a weird thing where in order for the API to work, we had to add async + awaits to all of the functions in the API that call `sendMessage`, but not to `sendMessage` itself. If we added async + await to sendMessage, we did not get the responses we expected in the `codap-plugin-starter-project `(see code implemented in the plugin [here](https://github.com/concord-consortium/codap-plugin-starter-project/blob/type-send-message/src/components/App.tsx#L46)).

### CODE REVIEW

File: src/api/codap-helper.ts [hunk 1]
```diff
@@ -28,20 +28,9 @@ const createMessage = (action: string, resource: string, values?: any) => {
...
-        reject("Request timeout");
-      } else if (result?.success) {
-        resolve(result.values);
-      } else {
-        const error_message = (result?.values?.error) ?? "unknown error";
-        reject(error_message);
-      }
-    });
-  });
+  return codapInterface.sendRequest(message, callback) as unknown as IResult;
```
Comment (kswenson) [inline src/api/codap-helper.ts:33]: I don't know how to weigh the equities here. If I were designing the API from scratch I would probably recommend something like this:
```typescript
const sendMessage = (action: Action, resource: string, values?: CodapItemValues) => {
  const message = createMessage(action, resource, values);
  return codapInterface.sendRequest(message) as unknown as IResult;
}
```
It's confusing to support returning a promise _and_ a callback in the same API.

But in an earlier version of the API it was apparently felt to be important to handle unpacking of the result inside `sendMessage`, presumably to simplify life for clients. I don't know to what extent that may still be true or to what extent compatibility with the earlier API should be a consideration here. My gut reaction is that this is a substantial change that should only be taken with careful consideration by stakeholders and not in a last-minute rush to publish the API.

File: src/api/codap-helper.ts [hunk 2]
```diff
@@ -58,14 +47,8 @@ export const initializePlugin = (options: IInitializePlugin) => {
...
-export const createTable = () => {
-  codapInterface.sendRequest({
-    action: "create",
-    resource: "component",
-    values: {
-      type: "caseTable"
-    }
-  });
+export const createTable = async () => {
+  return await sendMessage("create", "component", { type: "caseTable" });
```
Comment (kswenson) [inline src/api/codap-helper.ts:51]: You shouldn't need `async`/`await` here or in the other functions below that call `sendMessage`. This causes the code to wait here for the promise to be resolved, unpack the promise, and then package the result in a new promise to return to the client. Without the `async`/`await` the code just returns the original promise to the client.

>Also, there is a weird thing where in order for the API to work, we had to add async + awaits to all of the functions in the API that call `sendMessage`, but not to `sendMessage` itself. If we added async + await to sendMessage, we did not get the responses we expected in the `codap-plugin-starter-project `(see code implemented in the plugin [here](https://github.com/concord-consortium/codap-plugin-starter-project/blob/type-send-message/src/components/App.tsx#L46)).

I think this needs to be understood before this PR should be accepted.

**Review by kswenson — CHANGES_REQUESTED** [review]
I made some specific comments below, but this seems like substantial changes to the API and to the implementation that require more careful consideration. I'm not that familiar with the code and even less with the overall project goals and don't think it makes sense to rush to approve this for imminent publication without soliciting additional feedback.

File: src/api/codap-helper.ts (hunk 1 above)
Comment (lublagg) [reply src/api/codap-helper.ts:33]: @kswenson I think you're right that it is unnecessary to pass in the callback here. To my mind, the true purpose of the `sendMessage` function is to not have to write out the request message for every single function in the `codap-helpers` file. We aren't actually using a callback in any of these functions, so I think it's best to remove it. 

For a little more context -- the purpose of this new repository is to create a new npm library that would serve as the CODAP Plugin API. Currently, every existing plugin has essentially at least two files to help it interact with CODAP -- a straight copy of the `codapInterface.js` file, which never changes, and then some sort of helper file, to facilitate sending different types of requests or creating different types of listeners via the CODAP interface. So when it comes to `codap-helpers.js`, there is no true pre-existing API equivalent, in the sense that pretty much every currently-existing CODAP plugin has its own version of a `codap-helpers` file (often named connect.js, or codapConnect.js, or some such other). 

So, essentially, most plugins have essentially their 'own' API's that are already not using the older `sendMessage` function that you see -- which is from a [CODAP Plugin Starter Project](https://github.com/concord-consortium/codap-plugin-starter-project) that was started 5 years ago, and which we are now overhauling to be more up-to-date and useful. For reference, see the [NASS Plugin](https://github.com/concord-consortium/nass-plugin/tree/main/src/scripts) or the [Multidata Plugin](https://github.com/concord-consortium/multidata-plugin/tree/main/src/scripts). Note that neither use a version of the previous `sendMessage` function, but defines a bunch of helpers that use the `codapInterface.sendReqest` method.

File: src/api/codap-helper.ts (hunk 2 above)
Comment (lublagg) [reply src/api/codap-helper.ts:51]: @kswenson thank you Kirk -- I removed them, we just needed to have an async/await in the `sendMessage` function.

**Pull request merged**